scalding-core/src/test/scala/com/twitter/scalding/TypedSinkWithTypedImplementationTest.scala

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2012 Twitter, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+package com.twitter.scalding
+
+import cascading.flow.FlowDef
+import cascading.pipe.Pipe
+import cascading.tuple.Fields
+import org.scalatest.{Matchers, WordSpec}
+
+class TypedSinkWithTypedImplementation(path: String) extends TypedSink[String] {
+  private val fields = new Fields(0)
+
+  override def setter[U <: String]: TupleSetter[U] = TupleSetter.singleSetter[U]
+
+  override def writeFrom(pipe: Pipe)(implicit flowDef: FlowDef, mode: Mode): Pipe = {
+    TypedPipe.from[String](pipe, fields).write(TypedTsv[String](path))
+    pipe
```

Comment:
oh god.... why.........

Reply:
Actual use case is VKVS which aggregates values on write.